manifests/job/present.pp

```diff
@@ -7,7 +7,10 @@
 # Parameters:
 #
 #   config
-#     the content of the jenkins job config file (required)
+#     the content of the jenkins job config file (template based)
+#
+#   temp_config_path
```

Comment:
How about removing the 'temp_' prefix from the parameter name?  I think the name will confuse end users.